src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -312,6 +312,28 @@ async def pex_from_targets(request: PexFromTargetsRequest, python_setup: PythonS
             "`[python-setup].requirement_constraints` must also be set."
         )
 
+    if python_setup.lockfile:
+        # TODO(#12314): This does not handle the case where requirements are disjoint to the
+        #  lockfile. W/ constraints, we would avoid using a repository PEX in that case. We need
+        #  to decide what to do in a lockfile world, likely a) give up on repository PEX and
+        #  lockfile, or b) generate a new lockfile / error / warn to regenerate.
+        #
+        #  This will likely change when multiple lockfiles are supported. In the meantime, it
+        #  would be an issue because it would force you to have a single consistent resolve for
+        #  your whole repo.
```

Comment:
Not sure about the meantime, but: we should do `b)` when multiple resolves are supported. When the inputs to a resolve have changed, the resolve needs to re-run... and a requirement not existing in the lockfile means that the requirements have changed. So this is all part of the "detecting that we need to re-resolve" codepath.

Reply:
Good point, sounds good!